rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java

```diff
@@ -398,14 +417,44 @@ public void executeContextMenuEntry(String menuItem, String viewName, SWTLocator
 		try {
 			item = ContextMenuHelper.contextMenu(tree, menuItem.split("/"));
 		} catch (WidgetNotFoundException e) {
-			Thread.sleep(100);
-			logger.trace("Search menu entry a second time: {}", menuItem);
-			item = ContextMenuHelper.contextMenu(tree, menuItem.split("/"));
+			logger.error("exception during executeContextMenuEntry", e);
+			dumpMenuStatusFor(menuItem, viewName, locatorStrategy);
 		}
 		assertNotNull(item);
 		logger.trace("Click on menu item: {}", menuItem);
 		new SWTBotMenu(item).click();
 	}
+	
+	private void dumpMenuStatusFor(String menuItem, String viewName, ViewLocatorStrategy locatorStrategy) {
+		getDisplay().syncExec(new Runnable() {
+			@Override
+			public void run() {
+				try {
+					logger.debug("Search for tree in view = '{}' with strategy = '{}'", viewName,
+							locatorStrategy.name());
+					String[] menuItems = menuItem.split("/");
+					SWTBotTree tree = getTree(viewName, locatorStrategy);
+					logger.debug("Search context menu entry = '{}'", menuItem);
+					MenuItem item = null;
+					for (int i = 0; i < menuItems.length; i++) {
+						String[] subarray = Arrays.copyOf(menuItems, i + 1);
+						String reportPath = StringUtils.join(subarray, '/');
+						logger.debug("Searching for menupath ='{}'", reportPath);
+						item = ContextMenuHelper.contextMenu(tree, subarray);
+						MenuItem[] subitems = item.getMenu().getItems();
+						List<String> subitemsAsString = new ArrayList<>(subitems.length);
```

Comment:
So this list is just produced for output reasons? It would be nicer to use Java 8 features here:
```
String output = Arrays.stream(subitems).map(s -> s.getText()).collect(Collectors.joining(", "));
```